corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowserEditor.java

```diff
@@ -359,6 +359,74 @@ public int clearTable(String namespace, String tablename) {
         return -1;
     }
 
+    /**
+     * Add a record in a table and namespace
+     * @param namespace namespace of the table
+     * @param tableName name of the table
+     * @param newKey JSON string representing the key to add
+     * @param newRecord JSON string representing the value to add
+     * @param newMetadata JSON string representing the metadata to add
+     * @return CorfuDynamicRecord the newly added record.  null if no record
+     * was created
+     */
+    public CorfuDynamicRecord addRecord(String namespace, String tableName,
+                                        String newKey, String newRecord,
+                                        String newMetadata) {
+        System.out.println("\n======================\n");
+
+        TableName tableNameProto = TableName.newBuilder().setTableName(tableName)
+            .setNamespace(namespace).build();
+
+        Any defaultKeyAny =
+            dynamicProtobufSerializer.getCachedRegistryTable().get(tableNameProto)
+                .getPayload().getKey();
+        Any defaultValueAny =
+            dynamicProtobufSerializer.getCachedRegistryTable().get(tableNameProto)
+                .getPayload().getValue();
+        Any defaultMetadataAny =
+            dynamicProtobufSerializer.getCachedRegistryTable().get(tableNameProto)
+                .getPayload().getMetadata();
+
+        DynamicMessage newKeyMsg =
+            dynamicProtobufSerializer.createDynamicMessageFromJson(defaultKeyAny,
+                newKey);
+        DynamicMessage newValueMsg =
+            dynamicProtobufSerializer.createDynamicMessageFromJson(defaultValueAny,
+                newRecord);
+        DynamicMessage newMetadataMsg =
+            dynamicProtobufSerializer.createDynamicMessageFromJson(defaultMetadataAny,
+                newMetadata);
+
+        // Metadata can be empty or null but key or value should not
+        if (newKeyMsg == null || newValueMsg == null) {
+            log.error("New Key or Value message is null");
+            return null;
+        }
+
+        CorfuDynamicKey dynamicKey =
+            new CorfuDynamicKey(defaultKeyAny.getTypeUrl(), newKeyMsg);
+        CorfuDynamicRecord dynamicRecord =
+            new CorfuDynamicRecord(defaultValueAny.getTypeUrl(), newValueMsg,
+                defaultMetadataAny.getTypeUrl(), newMetadataMsg);
+
+        try {
+            CorfuTable<CorfuDynamicKey, CorfuDynamicRecord> table =
+                getTable(namespace, tableName);
```

Comment:
`getTable` will open a new table if it doesn't exist in objectCache. Is this expected behavior? same for editTable.

Reply:
getTable() does not use the OpenObject option so by default it would be NO_CACHE, which does not put the object in the cache if it does not already exist.  Am I missing something?

Reply:
By default the option is `CACHE`, see [code](https://github.com/CorfuDB/CorfuDB/blob/master/runtime/src/main/java/org/corfudb/runtime/view/SMRObject.java#L63)

Reply:
added a check based on the offline discussion